uint8 pin
State state
